### PR TITLE
Bugfix for fix mode of `AVOID_NULL_CHECK`

### LINKED ISSUE #1293: Bug in fix mode in rule `AVOID_NULL_CHECK`
## Describe the bug
Rule  `AVOID_NULL_CHECK` in fix mode give result 
```
splitOffset?.let {
   val listParentSearch = listOf(BINARY_EXPRESSION, PARENTHESIZED, FUN)
   if (parent.treeParent.elementType in listParentSearch || parent.treeParent.treeParent.elementType == FUNCTION_LITERAL) {
          parent = parent.treeParent
   } else {
            return checkBinaryExpression(parent, configuration)
   }
} ?: parent = parent.treeParent
```
But then an error is Variable Expected
## Expected behavior
```
splitOffset?.let {
    val listParentSearch = listOf(BINARY_EXPRESSION, PARENTHESIZED, FUN)
    if (parent.treeParent.elementType in listParentSearch || parent.treeParent.treeParent.elementType == FUNCTION_LITERAL) {
           parent = parent.treeParent
    } else {
           return checkBinaryExpression(parent, configuration) 
   }
 } ?: run { parent = parent.treeParent }
```
Code should be like this

## Environment information
* diktat version: 1.1.0

### PR DESCRIPTION
**Pull request by sanyavertolet**
This PR closes #1293
### What's done:
 - [x] Added isAssignment utility function to check if a line has assignment operator ('=') using simple regex
 - [x] Added a check for assignment operator when deciding if to put run block or not
 - [x] Added tests

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/StringUtils.kt [hunk 1]
```diff
@@ -37,6 +39,11 @@ val KOTLIN = KtTokens.KEYWORDS
  */
 val loggerPropertyRegex = "(?iu)^log(?:ger)?$".toRegex()
 
+/**
+ * @return true if this String includes assignment operator, false otherwise
+ */
+fun String.hasAssignment(): Boolean = contains(REGEX_FOR_ASSIGNMENT_DETECTION.toRegex())
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/StringUtils.kt:45]: Won't detection by ASTNode type be more reliable?

Comment (sanyavertolet) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/utils/StringUtils.kt]: Ok, replaced `hasAssignment` with `isAssignment` from `kotlin-compiler-embeddable`

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt [hunk 2]
```diff
@@ -138,26 +140,41 @@ class NullChecksRule(configRules: List<RulesConfig>) : DiktatRule(
...
+                    it.`else`
+                }
+            }
+            ?.blockExpressionsOrSingle()
+            ?.let { elements ->
+                elements.count() to elements.any { element ->
+                    KtPsiUtil.isAssignment(element)
+                }
+            }
+            ?: Pair(0, false)
```
Comment (0x6675636b796f75676974687562) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt:158]: I believe `0 to false` is more idiomatic.

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

**Review by petertrr — APPROVED** [review]

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt (hunk 2 above)
Comment (sanyavertolet) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter4/NullChecksRule.kt:158]: Well, it is either `(0 to false)` or `Pair(0, false)`. In this case second variant seem to be more clear

**Pull request merged**